File: lg/src/status.py
import lgpio
import time

SPI_BUS = 0
SPI_CS = 0
SPI_SPEED = 500000  # 500 kHz

# Define the command for reading the status
READ_STATUS_CMD = 0xC0  # Replace with the actual command from the datasheet

def send_command(handle, command):
    lgpio.spi_write(handle, [command])

def read_status(handle):
    send_command(handle, READ_STATUS_CMD)
    time.sleep(0.1)  # Wait for the command to process
    # Attempt to read 2 bytes instead of 1 for better status handling
    status, _ = lgpio.spi_read(handle, 2)  # Adjust number of bytes based on datasheet
    return status

def main():
    h = lgpio.spi_open(SPI_BUS, SPI_CS, SPI_SPEED)
    if h < 0:
        print("Failed to open SPI bus")
        return

    # Perform a reset or initialization if necessary
    # Example: send_command(h, INIT_CMD)  # Replace with actual initialization command

    status = read_status(h)
    print(f"Status: {status}")

    lgpio.spi_close(h)

if __name__ == "__main__":
    main()

File: lg/src/status.c
#include <stdio.h>
#include <stdlib.h>
#include <lgpio.h>
#include <unistd.h>

#define SPI_BUS 0
#define SPI_CS 0
#define SPI_SPEED 500000  // 500 kHz
#define READ_STATUS_CMD 0xC0  // Replace with the actual command from the datasheet

int init_spi() {
    int handle = lgGpiochipOpen(0);
    if (handle < 0) {
        fprintf(stderr, "Failed to open GPIO chip\n");
        return -1;
    }
    int spi_handle = lgSpiOpen(SPI_BUS, SPI_CS, SPI_SPEED, 0);
    if (spi_handle < 0) {
        fprintf(stderr, "Failed to open SPI bus\n");
        lgGpiochipClose(handle);
        return -1;
    }
    return spi_handle;
}

void send_command(int spi_handle, uint8_t command) {
    lgSpiWrite(spi_handle, &command, 1);
}

uint8_t* read_status(int spi_handle) {
    static uint8_t status[2];
    send_command(spi_handle, READ_STATUS_CMD);
    usleep(100000);  // Wait for the command to process
    lgSpiXfer(spi_handle, NULL, status, 2);  // Read two bytes of status
    return status;
}

int main() {
    int spi_handle = init_spi();
    if (spi_handle < 0) {
        return EXIT_FAILURE;
    }

    uint8_t* status = read_status(spi_handle);
    printf("Status: %02X %02X\n", status[0], status[1]);

    lgSpiClose(spi_handle);
    return EXIT_SUCCESS;
}

